troposphere/static/js/components/common/InstanceHistorySection.jsx

```diff
@@ -35,20 +46,71 @@ var InstanceHistorySection = React.createClass({
         });
     },
 
-    render: function() {
-        var instance = this.props.instance;
-        var content,
-            items,
-            deletedInfo;
+    style() {
+        return {
+            refreshIcon: {
+                float: "right",
+                color: "lightgrey"
+            },
+        }
+    },
 
-        if (instance.get("end_date")) {
-            deletedInfo = (
-                <lh>
-                    <strong>{"Deleted on " + moment(instance.get("end_date")).format("MMMM Do YYYY, h:mm a")}</strong>
-                </lh>
-            );
+    renderRefreshButton() {
+        let { refreshing } = this.state;
+        let { refreshIcon } = this.style();
+        let controlsClass = "glyphicon glyphicon-refresh";
+
+        if (refreshing) {
+            controlsClass += " refreshing"
+            refreshIcon.color = "inherit";
         }
-        if (!this.state.instanceHistory) {
+
+        return (
+        <span className={controlsClass} style={refreshIcon} onClick={this.onRefresh} />
+        );
+    },
+
+    renderHistoryRow: function(historyItem) {
+        let profile = stores.ProfileStore.get();
+        let is_staff_user = (profile) ? profile.get("is_staff") : false;
+        let extra = historyItem.get('extra'),
+            formattedStartDate = moment(historyItem.get("start_date")).format("MMMM Do YYYY, h:mm a"),
+            formattedEndDate = "Present";
+        if (historyItem.get("end_date") && historyItem.get("end_date").isValid()) {
+            formattedEndDate = moment(historyItem.get("end_date")).format("MMMM Do YYYY, h:mm a");
+        }
+        let formattedExtra;
+        let formattedExtraLines = [];
+        let show_traceback = (is_staff_user || context.hasEmulatedSession() );
+        if(extra && 'display_error' in extra) {
+            formattedExtra = (<p>{extra['display_error']}</p>);
+            if('traceback' in extra && show_traceback) {
+                let formattedText = extra['display_error'] + "\n" + extra['traceback'];
+                formattedExtra = (<CollapsibleOutput output={formattedText} />);
+            }
+        }
+        return (<tr key={historyItem.cid}>
+                    <td>{historyItem.get("status")}</td>
+                    <td>{formattedStartDate}</td>
+                    <td>{formattedEndDate}</td>
+                    <td>{formattedExtra}</td>
+                </tr>);
```

Comment:
@lenards I just saw this comment https://github.com/cyverse/troposphere/pull/672#discussion_r131512035

Would you like me to break this section out?

Reply:
Yes, I was thinking that you could pull this out. I think I have a provisional version of this, let me find a link.

Reply:
my component was/is missing a `key` attribute ... just an FYI

Reply:
@lenards Ok,  I've updated to include the `HistoryRow`. Can you re-review?